observatory-platform/observatory/platform/workflows/stream_telescope.py

```diff
@@ -268,19 +268,19 @@ def bq_delete_old(self, release: StreamRelease, **kwargs):
 
         dag_run: DagRun = kwargs["dag_run"]
         ti: TaskInstance = kwargs["ti"]
-        start_date = pendulum.instance(get_prev_start_date_success_task(dag_run, ti.task_id))
-        end_date = release.end_date
-        if (end_date - start_date).days + 1 >= self.bq_merge_days:
+        start_date = pendulum.instance(get_prev_start_date_success_task(dag_run, ti.task_id)).start_of("day")
+        now = pendulum.instance(ti.start_date).start_of("day")
+        if (now - start_date).days >= self.bq_merge_days:
```

Comment:
I personally think it would make things a lot simpler and easier to maintain to remove the merge days feature.